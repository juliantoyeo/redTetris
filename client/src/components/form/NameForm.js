import React from 'react'
import Button from '../subComponents/Button'

const styles = {
	form: {
		flexDirection: 'column',
		height: '15vw',
	},
}

const NameForm = (props) => {
	const { onSubmit, onChange } = props
	return (
		<div className={'formContainer'}>
			<span className={'header'}>Welcome to Red Tetris!</span>
			<form style={styles.form} onSubmit={onSubmit} key={'nameForm'} className={'standardForm'}>
				<label style={{ width: '70%' }}>
					Enter Your Name
				</label>
				<input
					name={'playerName'}
					className={'inputField'}
					style={{ width: '50%' }}
					type={'text'}
					onChange={(e) => onChange(e, 'playerName')}
					autoComplete='off'
					required
				/>
				<Button style={{ width: '15%' }} type={'submit'} text={'Play'} />
			</form>
		</div>
	)
}

export default NameForm