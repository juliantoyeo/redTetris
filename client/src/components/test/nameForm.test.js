import React from 'react';
import { shallow } from 'enzyme';

import NameForm from '../form/NameForm';

describe('Test NameForm component', () => {
	it('Should render properly', () => {
		const wrapper = shallow(<NameForm />);
		expect(wrapper).toMatchSnapshot();
	});
});