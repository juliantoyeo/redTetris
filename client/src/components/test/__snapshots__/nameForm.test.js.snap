// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NameForm component Should render properly 1`] = `
<div
  className="formContainer"
>
  <span
    className="header"
  >
    Welcome to Red Tetris!
  </span>
  <form
    className="standardForm"
    key="nameForm"
    style={
      Object {
        "flexDirection": "column",
        "height": "15vw",
      }
    }
  >
    <label
      style={
        Object {
          "width": "70%",
        }
      }
    >
      Enter Your Name
    </label>
    <input
      autoComplete="off"
      className="inputField"
      name="playerName"
      onChange={[Function]}
      required={true}
      style={
        Object {
          "width": "50%",
        }
      }
      type="text"
    />
    <Button
      style={
        Object {
          "width": "15%",
        }
      }
      text="Play"
      type="submit"
    />
  </form>
</div>
`;
